tests/Repository/AnnouncementRepositoryTest.php

```diff
@@ -128,6 +128,51 @@ public function dateTest(ObjectManager $manager): void
 
   }
 
+  /**
+   * testing user filter feature
+   * 
+   * @todo create 2 announcements with two different users, then test that the 
+   * created findByUser() function works properly by returning only the announcement(s) created with that user.
+   * 
+   * @author Daniel Boling 
+   */
+  public function userTest(ObjectManager $manager): void
+  {
+    $userRepo = static::getContainer()->get(UserRepository::class);
+    $testUser = $userRepo->findOneByUsername("david");
+
+    $announcement = new Announcement();
+    $announcement_date = new \DateTime('now');
+    $announcement->setSubject('autoSubject');
+    $announcement->setAuthor('autoAuthor');
+    $announcement->setUser($testUser);
+    $announcement->setDate($announcement_date);
+    $announcement->setText('autoText');
+    $manager->persist($announcement);
+
+    $testUser = $userRepo->findOneByUsername("dboling");
+
+    $announcement = new Announcement();
+    $announcement_date = new \DateTime('now');
+    $announcement->setSubject('autoSubject');
+    $announcement->setAuthor('autoAuthor');
+    $announcement->setUser($testUser);
+    $announcement->setDate($announcement_date);
+    $announcement->setText('autoText');
+    $manager->persist($announcement);
+
+    $testUser = $userRepo->findOneByUsername("david");
+    // Switching back to "cause possible issues"
+
+    $manager->flush();
+
+    $announcement = $this->entityManager
+      ->getRepository(AnnouncementRepository::class)
+      ->findByUser($testUser);
+
+    $this->assertSame('author', $announcement->getUser());
```

Comment:
Which announcement is this supposed to match?  None of the ones created in this test have an author of 'author'.  Is this matching something from one of the fixtures?  It would be better to check one of the announcements you created in this test.

Reply:
That was the idea, but now I'm not so sure why I put author into the first argument. This was written a couple of days ago and I just reworked it quickly to the new merges. The intent was to confirm that the announcement - pulled on line 169 - author matched the first test user. I put the back-and-forth in on purpose to prevent some false positive through caching or something.

Reply:
Fixed the assert to test against the testUser, which makes more sense - I think I wrote "author" in on accident when I literally thought "test against author".